tests/Microsoft.Bot.Connector.Tests/OAuthConnectorTests.cs

```diff
@@ -25,21 +31,42 @@ public void OAuthClient_ShouldThrowOnNullCredentials()
         }
 
         [Fact]
-        public async Task GetUserToken_ShouldThrowOnEmptyUserId()
+        public async Task GetTokenWithHttpMessagesAsync_ShouldThrowOnEmptyUserId()
         {
             var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
             await Assert.ThrowsAsync<ValidationException>(() => client.UserToken.GetTokenAsync(null, "mockConnection", string.Empty));
         }
 
         [Fact]
-        public async Task GetUserToken_ShouldThrowOnEmptyConnectionName()
+        public async Task GetTokenWithHttpMessagesAsync_ShouldThrowOnEmptyConnectionName()
         {
             var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("token"));
             await Assert.ThrowsAsync<ValidationException>(() => client.UserToken.GetTokenAsync("userid", null, string.Empty));
         }
 
+        [Fact]
+        public async Task GetTokenWithHttpMessagesAsync_ShouldThrowOnNoLocalBot()
+        {
+            var client = new OAuthClient(new Uri("http://localhost"), new BotAccessTokenStub("dummyToken"));
+            ServiceClientTracing.IsEnabled = true;
+
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("AGENT_OS")) &&
+                Environment.GetEnvironmentVariable("AGENT_OS").Equals("Windows_NT"))
+            {
+                // Automated Windows build exception:
+                await client.UserToken.GetTokenAsync("dummyUserid", "dummyConnectionName", "dummyChanneId", "dummyCode");
```

Comment:
```suggestion
                await client.UserToken.GetTokenAsync("dummyUserid", "dummyConnectionName", "dummyChannelId", "dummyCode");
```
All these can be properties to be reusable in the different tests.